lib/src/messages/server_messages.dart

```diff
@@ -69,6 +71,9 @@ class ParameterStatusMessage extends ServerMessage {
   factory ParameterStatusMessage.parse(PgByteDataReader reader) {
     final name = reader.readNullTerminatedString();
     final value = reader.readNullTerminatedString();
+    if (name.toLowerCase() == 'timezone') {
+      reader.timeZone.value = value;          
```

Comment:
Let's not update the `timeZone` value like this, it is a bit unexpected here. The `v3/connection.dart` `_handleMessage` already stores it in `PgConnectionImplementation._parameters` and we should just read that value from there. I think the settings should be immutable (maybe rename `TimeZoneSettings.value` into `defaultTimeZone`), and use the server-provided timeZone if present, otherwise the fallback.

Reply:
I believe that in this case it should not be immutable because when you use the SQL command "set timezone TO 'GMT';" you are changing the connection configuration, and this has to be reflected in the connection instance, because if after the user executes the SQL command to change the timezone and he reads the timezone property he will want to see the current value, right? I don't know the driver code very well, especially version 3, so I don't know if the driver user can read the timezone of the current connection through some method or property, as I have been very busy I haven't had much time to look at this, I don't know exactly how this could be done in another way.